packages/jest-mock/src/index.js

```diff
@@ -642,12 +642,22 @@ class ModuleMockerClass {
   }
 
   spyOn(object: any, methodName: any): any {
+    if (typeof object !== 'object' && typeof object !== 'function') {
+      throw new Error(
+        'Cannot spyOn on a primitive value; ' + this._typeOf(object) + ' given',
+      );
```

Comment:
Following on suggestion by Simen, here is a possibility:

```js
throw new Error(
  'Cannot spyOn on a primitive value\n' +
  printWithType('Got', object, printExpected),
);
```

Is it okay to add `jest-matcher-utils` to dependencies of this package?
```js
import {printExpected, printWithType} from 'jest-matcher-utils';
```

See for example:
* https://github.com/facebook/jest/blob/master/packages/expect/src/matchers.js#L579-L588
* https://github.com/facebook/jest/blob/master/packages/jest-matcher-utils/src/index.js#L123-L133

Reply:
Seems fine to me, these are just helpers 👍